app/models/widget.js

```diff
@@ -35,4 +35,29 @@ export default class Widget {
 
     db.write(records);
   }
+
+  // Validate for duplicate names
+  validateForDuplicateName() {
+    const records = db.read();
+    const existingName = records.find(r => r.name === this.name)
```

Comment:
although this app does not yet allow editing of widgets, this logic should still work if a user is updating an existing record.

Reply:
I may try to create an edit function to see if this duplicate name validator will work there as well.